src/components/CardDetails/CardDetails.tsx

```diff
@@ -0,0 +1,79 @@
+import React, {useEffect, useState} from "react";
+import {useParams} from "react-router-dom"
+
+import {Character} from "../Cards/CardsInterfaces";
+
+import s from "./CardDetails.module.css"
+
+
+const CardDetails = () => {
+    let {id} = useParams<string>();
```

Comment:
```suggestion
    let {id} = useParams();
```
по умолчанию итак string